lib/fluent/plugin_helper.rb

```diff
@@ -0,0 +1,36 @@
+#
+# Fluentd
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+require 'fluent/plugin_helper/event_emitter'
+require 'fluent/plugin_helper/thread'
+require 'fluent/plugin_helper/event_loop'
+require 'fluent/plugin_helper/timer'
+require 'fluent/plugin_helper/child_process'
+
+module Fluent
+  module PluginHelper
+    module Mixin
+      def self.included(mod)
+        mod.extend(Fluent::PluginHelper)
+      end
+    end
+
+    def helpers(*snake_case_symbols)
```

Comment:
Migrating some mixins in fluentd core to Plugin Helpers are not bad idea if it can be recreated with limited APIs and no implicit overriding of methods. I agree about your opinion that `Mixin` is too ruby-ish and to separate responsibilities to each helpers (with changing APIs from mixin to helpers).
Validating mechanism doesn't exist, because Plugin Helpers are only in Fluentd core.
I have no plan to prohibit to create any mixins (for 3rd party authors), but moving toward the direction without any mixins looks good especially in Fluentd core.